ports/esp32/modmachine.c

```diff
@@ -231,7 +231,7 @@ static mp_int_t mp_machine_reset_cause(void) {
 #endif
 
 NORETURN static void machine_bootloader_rtc(void) {
-    #if CONFIG_IDF_TARGET_ESP32S3
+    #if CONFIG_IDF_TARGET_ESP32S3 && MICROPY_HW_ENABLE_USBDEV
```

Comment:
I think it should be `MICROPY_HW_USB_CDC` because that's what the `usb_usj_mode()` function is guarded with in `usb.c`.